### ISSUE TITLE
Slower than classic Swift parson

### ISSUE THREAD

**Issue by penso** [orlandos-nl/IkigaJSON#21]
Hi,

Using the following:

```Swift
        let decoder = IkigaJSONDecoder()
        decoder.settings.dateDecodingStrategy = .formatted(ISO8601withFractionalSeconds())
        decoder.settings.nilValueDecodingStrategy = .decodeNilForKeyNotFound
        return decoder
```

with `ISO8601withFractionalSeconds` being:

```Swift
class ISO8601withFractionalSeconds: DateFormatter {
    static let iso8601withFractionalSeconds = ISO8601DateFormatter([.withInternetDateTime, .withFractionalSeconds])
    override func date(from string: String) -> Date? {
        iso8601withFractionalSeconds.date(from: string)
    }
}
```

I get about 3x slower performance than classic Swift. Is that expected?

**Comment by Joannis** [comment]
Hey! It's not expected, but I do know that @gwynne has a lot of changes coming up including performance improvements. I'll first await her PR before diving into this too deep. Though if you can write a simple unit test demonstrating your performance issue, do send it over!

**Comment by Joannis** [comment]
If you still remember; I'm curious what kind of dataset you were parsing.

**Comment by penso** [comment]
@Joannis it was an array of objects, maybe like 10k of them. Anything specific you're looking for? @jbl2024 made more tests within the same datatest and could give more details.

**Comment by aehlke** [comment]
End up having success with this?

**Comment by Joannis** [comment]
@aehlke it depends on your use case, but IkigaJSON nowadays starts a _little bit_ slower than Foundation, though using less memory for parsing the JSON. As your JSON blob grows, it should become faster and faster relative to Foundation.

Note that until the new Foundation arrives, it should always be faster than the iOS bundled JSONDecoder. If that's not the  case, let's be in touch.

**Comment by mkll** [comment]
@Joannis What about Linux' Foundation JSONDecoder comparing to IkigaJSON? Is it faster?

**Comment by Joannis** [comment]
@mkll only on very small datasets. So an object with 3 keys is faster in foundation on Linux.

**Comment by Joannis** [comment]
Also IkigaJSON saves almost all performance when working with data that you do not end up reading into Codable.

**Comment by mkll** [comment]
@Joannis So it makes sense to use IkigaJSON in Vapor for Content encoding/decoding?

**Comment by Joannis** [comment]
Yes @mkll, and IkigaJSON shares the underlying buffer type (ByteBuffer) for that reason

**Comment by Joannis** [comment]
While IkigaJSON supports iOS, it's mainly intended for Vapor users.
